src/foam/u2/crunch/wizardflow/SpinnerAgent.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.crunch.wizardflow',
+  name: 'SpinnerAgent',
+  documentation: `
+    Displays and exports a loading spinner to prevent a user from reloading the
+    page when data is still being saved after the wizard closes.
+  `,
+
+  implements: [
+    'foam.core.ContextAgent'
+  ],
+
+  imports: [
+    'ctrl'
+  ],
+
+  exports: [
+    'as spinnerAgent'
+  ],
+
+  requires: [
+    'foam.u2.LoadingSpinner',
+    'foam.u2.dialog.Popup'
+  ],
+
+  methods: [
+    async function execute() {
+      var popup = this.Popup.create({
+        closeable: false,
+      })
+        .start(this.LoadingSpinner)
+          .style({
+            display: 'flex',
+            width: '80px',
+            height: '80px'
+          })
+        .end();
+      this.ctrl.add(popup);
+      this.onDetach({
```

Comment:
You can just pass a function. Like this.onDetach(popup.close.bind(popup)) (unless close is a listener, and then just popup.close would work).